src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  // eslint-disable-next-line no-useless-constructor
+  constructor(name, health) {
+    super(name, health);
+  }
+
+  bite(target) {
+    if (target && target.hidden === false) {
```

Comment:
The `bite` method should check if the `target` is an instance of `Herbivore` and not hidden. Currently, it only checks if `target.hidden` is false without verifying if `target` is a `Herbivore`. This may lead to incorrect behavior when a `Carnivore` attempts to bite another `Carnivore` or an `Animal` that is not a `Herbivore`.